napi-inl.h

```diff
@@ -337,6 +339,15 @@ struct AccessorCallbackData {
   void* data;
 };
 
+// Debugging-purpose C++-style variant of sprintf().
```

Comment:
I would suggest not to use the variadic arguments here. They do not add any additional type safety in this context, but because this is a template we will have a copy of this function for each combination of parameter types. It would be better to use a simple C-style function instead. (I also propose to make the name less cryptic.)

```C++
inline std::string StringFormat(const char* format, ...)
{
	std::string result;
	va_list args;
	va_start(args, format);
	// stringLength does not include the null terminator.
	int stringLength = vsnprintf(nullptr, 0, format, args);
	// std::string adds 1 to the length to store the null terminator.
	result.resize(stringLength);
	vsnprintf(&result[0], stringLength + 1, format, args);
	va_end(args);
	return result;
}
```
I do not think that we need to check that the vsnprintf succeeded - `result.resize` will cause an exception or crash of the provided size is negative. If the expected size is zero, then we are safe because std::string automatically adds space for the null terminator.